Assets/MirageXR/DataModel/DataModel.cs

```diff
@@ -0,0 +1,319 @@
+﻿#nullable enable
+using System;
+using System.Collections.Generic;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+using UnityEngine;
+
+namespace DataModel
+{
+    [Serializable]
+    public class DataModel
+    {
+        [JsonProperty] public Guid Id { get; set; }
+        [JsonProperty] public DateTime CreationDate { get; set; }
+        [JsonProperty] public string Version { get; set; }
+    }
+
+    [Serializable]
+    public class Activity : DataModel
+    {
+        [JsonProperty] public User Owner { get; set; }
+        [JsonProperty] public string Name { get; set; }
+        [JsonProperty] public string Description { get; set; }
+        [JsonProperty] public string Language { get; set; } //is it needed?
+        [JsonProperty] public File Thumbnail { get; set; }
+        [JsonProperty] public Hierarchy Hierarchy { get; set; }
+        [JsonProperty] public List<Content> Content { get; set; } //todo
+    }
+
+    [Serializable]
+    public class Hierarchy
+    {
+        [JsonProperty] public List<HierarchyItem> Item { get; set; }
+    }
+
+    [Serializable]
+    public class HierarchyItem
+    {
+        [JsonProperty] public List<Guid>? StepIds { get; set; }
+        [JsonProperty] public List<HierarchyItem>? Item { get; set; }
+        [JsonProperty] public string Title { get; set; }
+        [JsonProperty] public string Description { get; set; }
+    }
+
+    [Serializable]
+    public class ActivityStep : DataModel
+    {
+        [JsonProperty] public uint Order { get; set; } //or it can be based on Activity.Steps?
+        [JsonProperty] public string Name { get; set; }
+        [JsonProperty] public string Description { get; set; }
+        [JsonProperty] public List<Guid> Contents { get; set; } //Content.Id
+        [JsonProperty] public Location Location { get; set; }
+        [JsonProperty] public List<Trigger>? Triggers { get; set; } // cannot be null?
+        [JsonProperty] public Comments Comments { get; set; }
+        [JsonProperty] public RequiredTools RequiredTools { get; set; }
+        [JsonProperty] public Media Media { get; set; }
+        [JsonProperty] public Comments PrivateNotes { get; set; }
+    }
+
+    [Serializable]
+    public class Comments
+    {
+        [JsonProperty] public List<Comment> Note { get; set; }
+    }
+
+    [Serializable]
+    public class Comment : DataModel
+    {
+        [JsonProperty] public User User { get; set; }
+        [JsonProperty] public string Text { get; set; }
+        [JsonProperty] public Media? Attachment { get; set; }
+    }
+
+    [Serializable]
+    public class Media
+    {
+        [JsonProperty] public List<File> File { get; set; } //one or many?
+    }
+
+    [Serializable]
+    public class RequiredTools
+    {
+        [JsonProperty] public List<string> Tools { get; set; }
+    }
+
+    [Serializable]
+    public class User : DataModel
+    {
+        [JsonProperty] public string Name { get; set; }
```

Comment:
I thought to use guid Id from DataModel

Reply:
I don't think that works, because what to do with downloaded units? The local guid assigned by a different user is not resolvable. We are changing login now to SSO on the identity server keycloak - we can try to bring the guid in line with the server uuid, but I think it is easier to resolve if this is the uuid, not the (local!) guid.